src/components/ConfigureSupersetEmbeddedDashboardModal/UpdateSupersetEmbeddedDashboard.js

```diff
@@ -0,0 +1,219 @@
+import { useDataMutation, useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    CircularLoader,
+    Modal,
+    ModalActions,
+    ModalContent,
+    ModalTitle,
+    NoticeBox,
+} from '@dhis2/ui'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+import { tFetchDashboards } from '../../actions/dashboards.js'
+import {
+    acClearSelected,
+    tSetSelectedDashboardById,
+} from '../../actions/selected.js'
+import { parseSupersetEmbeddedDashboardFieldValues } from '../../modules/parseSupersetEmbeddedDashboardFieldValues.js'
+import { useSupersetEmbeddedDashboardFieldsState } from '../../modules/useSupersetEmbeddedDashboardFieldsState.js'
+import { sGetSelectedId } from '../../reducers/selected.js'
+import styles from './styles/SupersetEmbeddedDashboardModal.module.css'
+import { SupersetEmbeddedDashboardFields } from './SupersetEmbeddedDashboardFields.js'
+
+const getDashboardQuery = {
+    dashboard: {
+        resource: 'dashboards',
+        id: ({ id }) => id,
+        params: {
+            fields: ['name', 'code', 'description', 'access', 'embedded[*]'],
+        },
+    },
+}
+const updateDashboardQuery = {
+    resource: 'dashboards',
+    type: 'update',
+    id: ({ id }) => id,
+    data: ({ values }) => parseSupersetEmbeddedDashboardFieldValues(values),
+    params: {
+        skipTranslation: true,
+        skipSharing: true,
+    },
+}
+const deleteDashboardQuery = {
+    resource: 'dashboards',
+    type: 'delete',
+    id: ({ id }) => id,
+}
+
+const parseErrorText = (error) =>
+    error?.details?.response?.errorReports[0]?.message ??
+    i18n.t('An unknown error occurred')
+
+export const UpdateSupersetEmbeddedDashboard = ({ closeModal }) => {
+    const dispatch = useDispatch()
+    const history = useHistory()
+    const id = useSelector(sGetSelectedId)
+    const {
+        loading: queryLoading,
+        error: queryError,
+        data: queryData,
+    } = useDataQuery(getDashboardQuery, { variables: { id } })
+    const dashboard = queryData?.dashboard
+    const name = dashboard?.name
+    const [mutationLoading, setMutationLoading] = useState(false)
+    const handeMutationError = useCallback(() => setMutationLoading(false), [])
+    const [updateDashboard, { error: updateError }] = useDataMutation(
+        updateDashboardQuery,
+        { variables: { id }, onError: handeMutationError }
+    )
+    const [deleteDashboard, { error: deleteError }] = useDataMutation(
+        deleteDashboardQuery,
+        { variables: { id }, onError: handeMutationError }
+    )
```

Comment:
I had to unset the loading state in case of errors in quite a peculiar way. A try/catch in `handleSubmit` and `handleDelete` would have been my preferred solution, but it turns out that the error is caught int the `useDataMutation` hook, so this was the only remaining solution unless we would switch to using the `dataEngine` directly.